### ISSUE TITLE
[C++20] [Modules] We can't diagnose the case that import before module declarations but not in GMF

### ISSUE THREAD

**Issue by ChuanqiXu9** [llvm/llvm-project#67627]
Reproducer:

```
// RUN: rm -rf %t
// RUN: mkdir -p %t
// RUN: split-file %s %t
//
// RUN: %clang_cc1 -std=c++20 %t/A.cppm -emit-module-interface -o %t/A.pcm
// RUN: %clang_cc1 -std=c++20 %t/B.cppm -fmodule-file=A=%t/A.pcm -fsyntax-only -verify

//--- A.cppm
export module A;

//--- B.cppm
import A;
export module B;
```

Actual behavior: no diagnostic emitted.
Expected behavior: an error diagnostic to tell we can't import A here.
